drivers/cdc_demo_client/cdc_demo.c

```diff
@@ -3,83 +3,234 @@
  *
  * cdc demo - Example of using: USB Communication Device Class
  *
- * Copyright 2019 Phoenix Systems
- * Author: Hubert Buczynski
+ * Copyright 2019, 2021 Phoenix Systems
+ * Author: Hubert Buczynski, Gerard Swiderski
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
 
-#include <errno.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
+#include <time.h>
 #include <unistd.h>
+#include <cdc_client.h>
 
-#include <sys/platform.h>
-#include <phoenix/arch/imxrt.h>
 
-#include <cdc_client.h>
+#define TIME_ONESECOND          1000000000ULL
+#define TIME_REALTIME_OVERFLOW  (TIME_ONESECOND / 5)
+
+#define USB_EP_BULK_EP          2
+#define USB_BUF_SIZE            0x1000
+
+#define LOG(str, ...)           do { if (1) fprintf(stderr, "cdc-demo: " str "\n", ##__VA_ARGS__); } while (0)
+#define LOG_ERROR(str, ...)     do { fprintf(stderr, __FILE__  ":%d error: " str "\n", __LINE__, ##__VA_ARGS__); } while (0)
+
+
+typedef struct {
+	struct {
+		void *ptr;
+		size_t size;
+	} buf;
 
-#define RCV_MODE 2
-#define SENDING_MODE 1
+	struct {
+		unsigned int steps;
+		uint64_t current;
+		uint64_t schedule;
+		uint64_t start;
+	} time;
 
-#define CDC_MODE RCV_MODE
+	struct {
+		size_t totalSent;
+	} stat;
```

Comment:
Is it necessary to create struct for one variable?

Reply:
The first (unpublished) version of this demo had other stat counters, `totalReceived`, `badCrc`, etc.  Because the sub-structure is there `sizeof(size_t)`,  I left it as is for readability.

What **I would complain about** is the sub-structure `flags` below with **the bitfields** because of `uint8_t` it will surely get padding there.